### ISSUE TITLE
feature request: Patch "all containers" or regex

### ISSUE THREAD

**Issue by flbla** [bohlenc/k8s-pod-mutator-webhook#3]
I would like to be able to patch all containers without knowing there name (or with a regex)

my use case is to be able to copy data from a init container to all containers in pod without knowing name of containers:
```
webhook:
  patch: |
    spec:
      initContainers:
      - name: ca
        image: alpine
        command: ["sh", "-c", "cp -r /etc/ssl/certs /volume"]
        volumeMounts:
        - mountPath: /volume
          name: cacerts
      containers:
      - name: *
        volumeMounts:
        - mountPath: /etc/ssl
          name: cacerts
      volumes:
      - name: cacerts
        emptyDir: {}
```

I tried with a "*" it doesn't work

**Comment by bohlenc** [comment]
Hi @flbla 
Interesting use case! Have you tried omitting "- name" completely, like so:

```
- volumeMounts:
  - mountPath: /etc/ssl
    name: cacerts
```

I will have a shot at it too. I'll come back to you later!

**Comment by flbla** [comment]
Hi @bohlenc 
yes, it failed with an error:

time="2021-03-05T13:17:55Z" level=error msg="could not create json patch: could not apply strategic merge patch: map: map[volumeMounts:[map[mountPath:/etc/ssl name:cacerts]]] does not contain declared merge key: name"

```
webhook:
  patch: |
    spec:
      initContainers:
      - name: ca
        image: alpine
        command: ["sh", "-c", "cp -r /etc/ssl/certs /volume"]
        volumeMounts:
        - mountPath: /volume
          name: cacerts
      containers:
      - volumeMounts:
        - mountPath: /etc/ssl
          name: cacerts
      volumes:
      - name: cacerts
        emptyDir: {}
```

thanks! :)

**Comment by bohlenc** [comment]
Yeah, forgot that "name" is the merge key for containers.

I think about adding something alone these lines: if a field that is declared as "merge key" is specified as "*" in a patch -> patch all available elements. That would cover your specific use-case but would also be broadly applicable to many other scenarios, e.g. adding to volumes etc.

**Comment by bohlenc** [comment]
I just merged code (see #4) to support this use-case. Let me know if that works for you.

**Comment by flbla** [comment]
it works perfectly :)
thank you @bohlenc
